src/registry.rs

```diff
@@ -102,7 +104,11 @@ impl Registry {
 
         for (index, worker) in workers.into_iter().enumerate() {
             let registry = registry.clone();
-            thread::spawn(move || unsafe { main_loop(worker, registry, index) });
+            let mut b = thread::Builder::new();
+            if let Some(name) = configuration.base_thread_name(index) {
+                b = b.name(name);
+            }
+            b.spawn(move || unsafe { main_loop(worker, registry, index) });
```

Comment:
Will simply `.unwrap()` the call suffice? Failing to spawn a thread seems a serious enough issue to shut down everything to me. 

I haven't really understood the initialization routine of rayon yet; is the initialization performed in the main thread, or simply in the thread first to call eg. `.par_iter()`, such that a `panic` in the init logic will cause only that thread to panic?